src/rgen/Microsoft.Macios.Generator/DataModel/BindingInfo.cs

```diff
@@ -8,7 +8,7 @@ namespace Microsoft.Macios.Generator.DataModel;
 /// This struct works as a union to store the possible BindingTypeData that can be present in the bindings.
 /// </summary>
 [StructLayout (LayoutKind.Explicit)]
-readonly struct BindingData : IEquatable<BindingData> {
+readonly struct BindingInfo : IEquatable<BindingInfo> {
 	// make the struct smaller by making all wrapped structs be at index 0
 	[FieldOffset (0)] readonly BindingType bindingType;
 	[FieldOffset (8)] readonly BindingTypeData bindingTypeData;
```

Comment:
Question - since this is making an assumption about the size of `BindingType`, will there be a possibility where this will either (a) break on a different platform or (b) produce undesirable results if the definition of `BindingType` changed? Would it make sense to make that second FieldOffset to be `[FieldOffset (sizeof(nint))]` (or your other favorite pointer sized type)